### PR TITLE
order input for snapshot-match to fix reference-replacement

### PR DESCRIPTION
**Pull request by steffyP**
Fix problem with reference-replacement for snapshots: match-object needs to be sorted to guarantee same outcome/replacement order (as tested/suggested by @bentsku).

### CODE REVIEW

**Review by bentsku — APPROVED** [review]
LGTM! I don't know how we could do this better, as it might still have chances to fail (we can imagine a resource in the AWS response that we don't return, which would increment the number), but I don't have a better idea right now! Thanks a lot for jumping on this issue! 🙏

**Review by dominikschubert — APPROVED** [review]
LGTM :+1: 

An alternative I could think of would involve more the comparison side of the snapshot system.

Basically all token qualifiers (e.g. the `1` in `<resource:1>`) could be compared internally in a single test run while the `resource` part is compared between saved & executed test.

The drawback of this would be we the fact that the comparisons won't be a "pure" JSON comparison, so I think for now this approach in the PR makes more sense :+1:

**Comment by steffyP** [comment]
updated the dict-ordering once more to include deep-ordering of dicts as requested by @bentsku 

/cc @dominikschubert

File: localstack/testing/snapshots/prototype.py [hunk 1]
```diff
@@ -138,9 +138,11 @@ def match(self, key: str, obj: dict) -> None:
 
         self.called_keys.add(key)
 
-        self.observed_state[
-            key
-        ] = obj  # TODO: track them separately since the transformation is now done *just* before asserting
+        # order the obj to guarantee reference replacement works as expected
+        self.observed_state[key] = (
+            self._order_dict(obj) if isinstance(obj, dict) else obj
```
Comment (bentsku) [inline localstack/testing/snapshots/prototype.py:143]: nit: I think you don't need the first `isinstance` anymore as it checks it in the function now

**Review by bentsku — APPROVED** [review]
LGTM! I hope this will be enough for now, I think it will! The idea of Dominik is great as well, can't wait to see how the snapshots will help us do our job better, and how they will improve over time! 
I'll reiterate, but I really love the snapshots mechanism! ❤️

**Pull request merged**